From 1bb940af5f41ffe108de812067b5d4c3d69f8a43 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Tue, 24 Sep 2019 21:39:07 +0200
Subject: [PATCH] uboot-tools: Add additional tools and refactor
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This cleans original uboot-envtools package. The original hacky approach
to compilation was dropped in favor of standard configuration usage.

There seems to be no need for fstools as a dependency.

The patch is not required. There is standard approach for
crosscompilation.

The binaries can be shared between boards. There is no board specific
configuration used in this package.

Signed-off-by: Karel Kočí <karel.koci@nic.cz>
---
 .../uboot-envtools/patches/001-compile.patch  | 13 ----
 .../{uboot-envtools => uboot-tools}/Makefile  | 59 ++++++++++++-------
 .../files/apm821xx                            |  0
 .../files/ar71xx                              |  0
 .../files/ath79                               |  0
 .../files/cns3xxx                             |  0
 .../files/imx6                                |  0
 .../files/ipq40xx                             |  0
 .../files/ipq806x                             |  0
 .../files/kirkwood                            |  0
 .../files/lantiq                              |  0
 .../files/layerscape                          |  0
 .../files/mpc85xx                             |  0
 .../files/mvebu                               |  0
 .../{uboot-envtools => uboot-tools}/files/mxs |  0
 .../files/oxnas                               |  0
 .../files/pistachio                           |  0
 .../files/ramips                              |  0
 .../files/uboot-envtools.sh                   |  0
 19 files changed, 37 insertions(+), 35 deletions(-)
 delete mode 100644 package/boot/uboot-envtools/patches/001-compile.patch
 rename package/boot/{uboot-envtools => uboot-tools}/Makefile (57%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/apm821xx (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/ar71xx (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/ath79 (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/cns3xxx (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/imx6 (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/ipq40xx (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/ipq806x (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/kirkwood (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/lantiq (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/layerscape (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/mpc85xx (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/mvebu (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/mxs (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/oxnas (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/pistachio (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/ramips (100%)
 rename package/boot/{uboot-envtools => uboot-tools}/files/uboot-envtools.sh (100%)

diff --git a/package/boot/uboot-envtools/patches/001-compile.patch b/package/boot/uboot-envtools/patches/001-compile.patch
deleted file mode 100644
index 34072f3..0000000
--- a/package/boot/uboot-envtools/patches/001-compile.patch
+++ /dev/null
@@ -1,13 +0,0 @@
---- a/tools/env/Makefile
-+++ b/tools/env/Makefile
-@@ -8,6 +8,10 @@
- # with "CC" here for the maximum code reuse of scripts/Makefile.host.
- override HOSTCC = $(CC)
- 
-+ifneq ($(TARGET_CFLAGS),)
-+HOSTCFLAGS = $(TARGET_CFLAGS)
-+endif
-+
- # Compile for a hosted environment on the target
- HOST_EXTRACFLAGS  = -I$(srctree)/tools \
- 		$(patsubst -I%,-idirafter%, $(filter -I%, $(UBOOTINCLUDE))) \
diff --git a/package/boot/uboot-envtools/Makefile b/package/boot/uboot-tools/Makefile
similarity index 57%
rename from package/boot/uboot-envtools/Makefile
rename to package/boot/uboot-tools/Makefile
index 4be043d..e4ea11c 100644
--- a/package/boot/uboot-envtools/Makefile
+++ b/package/boot/uboot-tools/Makefile
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2006-2014 OpenWrt.org
+# Copyright (C) 2006-2019 OpenWrt.org
 #
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
@@ -7,7 +7,7 @@
 
 include $(TOPDIR)/rules.mk
 
-PKG_NAME:=uboot-envtools
+PKG_NAME:=uboot-tools
 PKG_DISTNAME:=u-boot
 PKG_VERSION:=2019.07
 PKG_RELEASE:=1
@@ -20,46 +20,47 @@ PKG_SOURCE_URL:=https://git.denx.de/u-boot.git
 PKG_SOURCE_VERSION:=e5aee22e4be75e75a854ab64503fc80598bc2004
 PKG_MIRROR_HASH:=58c1ecaf901b6bf65c5e872b5449b642694ae5acebf61f91f0d4bc20b4c654b7
 
-PKG_BUILD_DEPENDS:=fstools
-
 PKG_LICENSE:=GPL-2.0 GPL-2.0+
 PKG_LICENSE_FILES:=Licenses/README
 
-PKG_FLAGS:=nonshared
-
-PKG_BUILD_PARALLEL:=1
-
 include $(INCLUDE_DIR)/package.mk
 
-define Package/uboot-envtools
+define Package/Common
   SECTION:=utils
   CATEGORY:=Utilities
   SUBMENU:=Boot Loaders
-  TITLE:=read/modify U-Boot bootloader environment
   URL:=http://www.denx.de/wiki/U-Boot
 endef
 
+define Package/uboot-envtools
+  $(call Package/Common)
+  TITLE:=read/modify U-Boot bootloader environment
+endef
+
 define Package/uboot-envtools/description
  This package includes tools to read and modify U-Boot bootloader environment.
 endef
 
-define Build/Configure
-	touch $(PKG_BUILD_DIR)/include/config.h
-	mkdir -p $(PKG_BUILD_DIR)/include/config
-	touch $(PKG_BUILD_DIR)/include/config/auto.conf
-	mkdir -p $(PKG_BUILD_DIR)/include/generated
-	touch $(PKG_BUILD_DIR)/include/generated/autoconf.h
+define Package/uboot-tools
+  $(call Package/Common)
+  TITLE:=U-Boot tools for image creation and more
+  DEPENDS:=+libopenssl
+  PROVIDES:=uboot-mkimage
 endef
 
-TARGET_CFLAGS += -I$(STAGING_DIR)/usr/include
+define Package/uboot-envtools/description
+ This package includes tools to create U-Boot bootloader images.
+endef
+
+define Build/Configure
+	$(call Build/Compile/Default,tools-only_defconfig)
+endef
 
 define Build/Compile
-	$(MAKE) -C $(PKG_BUILD_DIR) \
+	$(call Build/Compile/Default, \
 		CROSS_COMPILE="$(TARGET_CROSS)" \
-		TARGET_CFLAGS="$(TARGET_CFLAGS)" \
-		HOSTLDFLAGS= \
-		no-dot-config-targets=envtools \
-		envtools
+		CROSS_BUILD_TOOLS=y \
+		envtools tools-only)
 endef
 
 define Package/uboot-envtools/conffiles
@@ -80,4 +81,18 @@ define Package/uboot-envtools/install
 	)
 endef
 
+define Package/uboot-tools/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/dumpimage $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/fdtgrep $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/fit_check_sign $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/fit_info $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/gen_eth_addr $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/gen_ethaddr_crc $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/img2srec $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/mkenvimage $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/mkimage $(1)/usr/sbin
+endef
+
 $(eval $(call BuildPackage,uboot-envtools))
+$(eval $(call BuildPackage,uboot-tools))
diff --git a/package/boot/uboot-envtools/files/apm821xx b/package/boot/uboot-tools/files/apm821xx
similarity index 100%
rename from package/boot/uboot-envtools/files/apm821xx
rename to package/boot/uboot-tools/files/apm821xx
diff --git a/package/boot/uboot-envtools/files/ar71xx b/package/boot/uboot-tools/files/ar71xx
similarity index 100%
rename from package/boot/uboot-envtools/files/ar71xx
rename to package/boot/uboot-tools/files/ar71xx
diff --git a/package/boot/uboot-envtools/files/ath79 b/package/boot/uboot-tools/files/ath79
similarity index 100%
rename from package/boot/uboot-envtools/files/ath79
rename to package/boot/uboot-tools/files/ath79
diff --git a/package/boot/uboot-envtools/files/cns3xxx b/package/boot/uboot-tools/files/cns3xxx
similarity index 100%
rename from package/boot/uboot-envtools/files/cns3xxx
rename to package/boot/uboot-tools/files/cns3xxx
diff --git a/package/boot/uboot-envtools/files/imx6 b/package/boot/uboot-tools/files/imx6
similarity index 100%
rename from package/boot/uboot-envtools/files/imx6
rename to package/boot/uboot-tools/files/imx6
diff --git a/package/boot/uboot-envtools/files/ipq40xx b/package/boot/uboot-tools/files/ipq40xx
similarity index 100%
rename from package/boot/uboot-envtools/files/ipq40xx
rename to package/boot/uboot-tools/files/ipq40xx
diff --git a/package/boot/uboot-envtools/files/ipq806x b/package/boot/uboot-tools/files/ipq806x
similarity index 100%
rename from package/boot/uboot-envtools/files/ipq806x
rename to package/boot/uboot-tools/files/ipq806x
diff --git a/package/boot/uboot-envtools/files/kirkwood b/package/boot/uboot-tools/files/kirkwood
similarity index 100%
rename from package/boot/uboot-envtools/files/kirkwood
rename to package/boot/uboot-tools/files/kirkwood
diff --git a/package/boot/uboot-envtools/files/lantiq b/package/boot/uboot-tools/files/lantiq
similarity index 100%
rename from package/boot/uboot-envtools/files/lantiq
rename to package/boot/uboot-tools/files/lantiq
diff --git a/package/boot/uboot-envtools/files/layerscape b/package/boot/uboot-tools/files/layerscape
similarity index 100%
rename from package/boot/uboot-envtools/files/layerscape
rename to package/boot/uboot-tools/files/layerscape
diff --git a/package/boot/uboot-envtools/files/mpc85xx b/package/boot/uboot-tools/files/mpc85xx
similarity index 100%
rename from package/boot/uboot-envtools/files/mpc85xx
rename to package/boot/uboot-tools/files/mpc85xx
diff --git a/package/boot/uboot-envtools/files/mvebu b/package/boot/uboot-tools/files/mvebu
similarity index 100%
rename from package/boot/uboot-envtools/files/mvebu
rename to package/boot/uboot-tools/files/mvebu
diff --git a/package/boot/uboot-envtools/files/mxs b/package/boot/uboot-tools/files/mxs
similarity index 100%
rename from package/boot/uboot-envtools/files/mxs
rename to package/boot/uboot-tools/files/mxs
diff --git a/package/boot/uboot-envtools/files/oxnas b/package/boot/uboot-tools/files/oxnas
similarity index 100%
rename from package/boot/uboot-envtools/files/oxnas
rename to package/boot/uboot-tools/files/oxnas
diff --git a/package/boot/uboot-envtools/files/pistachio b/package/boot/uboot-tools/files/pistachio
similarity index 100%
rename from package/boot/uboot-envtools/files/pistachio
rename to package/boot/uboot-tools/files/pistachio
diff --git a/package/boot/uboot-envtools/files/ramips b/package/boot/uboot-tools/files/ramips
similarity index 100%
rename from package/boot/uboot-envtools/files/ramips
rename to package/boot/uboot-tools/files/ramips
diff --git a/package/boot/uboot-envtools/files/uboot-envtools.sh b/package/boot/uboot-tools/files/uboot-envtools.sh
similarity index 100%
rename from package/boot/uboot-envtools/files/uboot-envtools.sh
rename to package/boot/uboot-tools/files/uboot-envtools.sh
-- 
2.23.0

